File: src/MultiAgentApp/Configuration/TelemetryOptions.cs
namespace MultiAgentApp.Configuration;

/// <summary>
/// Telemetry options.
/// Application Insights connection is optional today; set the connection string once
/// the Azure environment is available.
/// </summary>
public sealed class TelemetryOptions
{
    public const string SectionName = "Telemetry";

    /// <summary>
    /// Application Insights connection string.
    /// When non-empty, telemetry is sent to Azure Monitor Application Insights.
    /// Retrieve from the Azure portal or Key Vault; do not hard-code in source.
    /// </summary>
    public string ApplicationInsightsConnectionString { get; set; } = string.Empty;

    /// <summary>
    /// When <c>true</c>, detailed Microsoft Agent Framework activity traces are emitted
    /// (tool calls, token counts, latencies). Disable in production to reduce costs.
    /// </summary>
    public bool EnableDetailedTracing { get; set; } = true;
}

File: src/multi_agent_app/config/telemetry_options.py
"""Configuration dataclasses for telemetry / OpenTelemetry settings."""

from __future__ import annotations
from dataclasses import dataclass, field


@dataclass
class TelemetryOptions:
    """Telemetry options.

    Application Insights connection is optional today; set the connection string
    once the Azure environment is available.
    """

    SECTION_NAME: str = field(
        default="Telemetry", init=False, repr=False, compare=False
    )

    application_insights_connection_string: str = ""
    """When non-empty, telemetry is sent to Azure Monitor Application Insights."""

    enable_detailed_tracing: bool = True
    """When True, detailed tool-call and orchestration traces are emitted.
    Disable in production to reduce costs."""

    @classmethod
    def from_dict(cls, d: dict) -> TelemetryOptions:
        return cls(
            application_insights_connection_string=d.get(
                "ApplicationInsightsConnectionString", ""
            ),
            enable_detailed_tracing=d.get("EnableDetailedTracing", True),
        )
